docs/content/tutorial/connecting-to-external-services.mdx

```diff
@@ -0,0 +1,186 @@
+---
+title: "Tutorial, part seven: Connecting to external services | Dagster Docs"
+description: Learn how to connect to external services in Dagster with resources
+---
+
+Throughout the tutorial, you have built a data pipeline that connects to different external services, such as Hacker News. By building this asset graph, Dagster acts as a control plane of multiple different external services. However, connecting to these services and managing their use can be difficult. Resources are the recommended way to manage these connections in Dagster.
+
+Resources are Dagster objects that connect to external services, such as Slack, Snowflake, or S3. In this section, you’ll learn more about resources, including how to:
+
+- Define a resource and use it in an asset
+- Configure a resource
+- Customize a resource based on the environment where Dagster is running
+
+## Why use resources?
+
+In the `topstory_ids` and `topstories` assets, you used the `requests` library to communicate with the Hacker News API. This code gets harder to maintain if:
+
+- Connections need to be shared across multiple assets
+- You need to authenticate with the external service
+- Configurations need to be consistent across its usage
+- If any of the above changes based on environment (e.g. development vs. production)
```

Comment:
may worth calling out something like vary tokens in dev vs prod to avoid rate limit. and may also worth saying something about tests as resources play an important role in unit tests.